redaxo/src/core/lib/login/user.php

```diff
@@ -50,7 +50,28 @@ public static function get(int $id): ?self
         return static::getInstance($id, static function (int $id) {
             $sql = rex_sql::factory()->setQuery('SELECT * FROM '.rex::getTable('user').' WHERE id = ?', [$id]);
 
-            return $sql->getRows() ? new static($sql) : null;
+            if ($sql->getRows()) {
+                $user = new static($sql);
+                static::addInstance('login_' . $user->getLogin(), $user);
+                return $user;
+            }
+
+            return null;
+        });
+    }
+
+    public static function getByLogin(string $login): ?self
```

Comment:
In R4 hatten wir einige solcher `getByX`-Methoden, in R5 hatten wir die aber abgeschafft (da an den alten Stellen nicht mehr benötigt). Ich meine wir haben hier im Repo keine einzige `getByX`-Methode mehr.
Zuletzt haben wir Vergleichbares aber an anderer Stelle auch wieder eingeführt, und zwar `rex_template::forKey()` und `rex_module::forKey()`.
Daher überlege ich, ob wir hier dann nicht auch eher dem neuen Namenschema folgen sollten, also `rex_user::forLogin()`?

(Eigentlich gefällt mir für solche Methoden am besten das Schema `fromKey()` und `fromLogin()` (was ich auch vermehrt in anderen Libs sehe). Aber da wir nun schon `forKey()` haben, passt `forLogin()` wohl besser.